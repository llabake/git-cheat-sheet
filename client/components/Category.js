import React from 'react';

import Cheat from './Cheat';

const Category = ({ category }) => (
  <div className="category col s4">
    <span className="category-title">{category.title}</span>
    <div className="category-cheat">
      {
        category.cheats.map(cheat => <Cheat key={cheat._id} cheat={cheat} />)
      }
    </div>
  </div>
);

export default Category;