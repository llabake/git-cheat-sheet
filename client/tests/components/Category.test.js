import React from 'react';
import { shallow } from 'enzyme';

import Category from '../../components/Category';

const setup = () => {
  const props = {
    category: {
      cheats: [{}],
    },
  };

  const wrapper = shallow(<Category {...props} />);

  return { wrapper, props };
};

describe('<Category />', () => {
  const { wrapper } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
