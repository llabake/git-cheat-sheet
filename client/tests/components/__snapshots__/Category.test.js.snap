// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Category /> should render correctly 1`] = `
<div
  className="category col s4"
>
  <span
    className="category-title"
  />
  <div
    className="category-cheat"
  >
    <Cheat
      cheat={Object {}}
    />
  </div>
</div>
`;
